### PR TITLE
[INDY-2010] add command GET_AUTH_RULE

### PR DESCRIPTION
**Pull request by Toktar**
Wait https://github.com/hyperledger/indy-plenum/pull/1121

### CODE REVIEW

File: indy_common/types.py [hunk 1]
```diff
@@ -368,6 +368,19 @@ class ClientAuthRuleOperation(MessageValidator):
     )
 
 
+class ClientGetAuthRuleOperation(MessageValidator):
+    schema = (
+        (TXN_TYPE, ConstantField(GET_AUTH_RULE)),
+        (AUTH_ACTION, ChooseField(values=(ADD_PREFIX, EDIT_PREFIX), optional=True)),
+        (AUTH_TYPE, LimitedLengthStringField(max_length=NAME_FIELD_LIMIT, optional=True)),
+        (FIELD, LimitedLengthStringField(max_length=NAME_FIELD_LIMIT, optional=True)),
+        (OLD_VALUE, LimitedLengthStringField(max_length=NAME_FIELD_LIMIT,
```
Comment (ashcherbakov) [inline indy_common/types.py]: OLD_VALUE can be empty now (see https://github.com/hyperledger/indy-node/pull/1202)

File: indy_node/server/config_req_handler.py [hunk 2]
```diff
@@ -60,14 +65,16 @@ def _doStaticValidationAuthRule(self, identifier, reqId, operation):
...
-        if auth_key not in self.write_req_validator.auth_map and \
-                auth_key not in self.write_req_validator.anyone_can_write_map:
-            raise InvalidClientRequest(identifier, reqId,
-                                       "Unknown authorization rule: key '{}' is not "
-                                       "found in authorization map".format(auth_key))
+    def _doStaticValidationGetAuthRule(self, identifier, req_id, operation):
+        required_fields = list(dict(ClientGetAuthRuleOperation.schema).keys())
+        required_fields.remove(OLD_VALUE)
+        if len(operation) > 1:
+            if not set(required_fields).issubset(set(operation.keys())):
```
Comment (ashcherbakov) [inline indy_node/server/config_req_handler.py]: Should we do similar checks as in `_doStaticValidationAuthRule` related to `OLD_VALUE `?

File: indy_node/server/request_handlers/read_req_handlers/get_auth_rule_handler.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                                  data=data,
+                                  last_seq_no=seq_no,
+                                  update_time=update_time,
+                                  proof=proof)
+
+        result.update(request.operation)
+        return result
+
+    def _get_auth_map(self):
+        data, seq_no, update_time, proof = None, None, None, None
```
Comment (ashcherbakov) [inline indy_node/server/request_handlers/read_req_handlers/get_auth_rule_handler.py]: Is it WIP?

File: docs/source/requests.md [hunk 4]
```diff
@@ -1541,6 +1542,159 @@ A generic request to get a transaction from Ledger by its sequence number.
 }
 ```
 
+### GET_AUTH_RULE
+
+A request to get a constraint for an authentication rule or a full list of rules from Ledger by the auth key parameters.
+Two options are possible in a request build:
+- If the request has a full list of parameters (or without `old_value`), then the reply will contain one constraint for this key.
+- If the request are not contain fields other than txn_type, the response will contain a full list of authentication rules.
```
Comment (ashcherbakov) [inline docs/source/requests.md]: does not contain

File: indy_node/server/config_req_handler.py (hunk 2 above)
Comment (Toktar) [reply indy_node/server/config_req_handler.py]: I think, no. It's not a command which change something. If a request contains excess fields, but we can found a key maybe will be better to do it.

**Comment by ashcherbakov** [comment]
(ci) test this please

**Comment by Toktar** [comment]
(ci) test this please

**Pull request merged**